app/models/concerns/availability_status_concern.rb

```diff
@@ -33,4 +33,17 @@ def reset_availability!
     reset_availability
     save!
   end
+
+  # sets availability status from caller's status
+  # now Application -> Source
+  #
+  # @param caller [ActiveRecord::Base]
+  def set_availability!(caller)
```

Comment:
https://apidock.com/ruby/v2_6_3/Kernel/caller is also method, I am thinking whether it is not confusing.

Reply:
maybe, do you have some suggestion about other name?

Reply:
It looks that `caller` is object or  resource so maybe it can be `caller_resource` or `caller_object` or just `resource`.